### ISSUE TITLE
Blacklists too strict

### ISSUE THREAD

**Issue by myfirstnameispaul** [mail-in-a-box/mailinabox#1638]
In signing up for the Dovecot mailing list, MiaB reports the following in the logs:

````
Sep  4 21:20:47 mail postfix/smtpd[21322]: NOQUEUE: reject: RCPT from talvi.dovecot.org[2a04:3542:1000:910:acc1:5bff:fe5e:125c]: 554 5.7.1 Service unavailable; Client host [2a04:3542:1000:910:acc1:5bff:fe5e:125c] blocked using zen.spamhaus.org; https://www.spamhaus.org/sbl/query/SBLCSS; from=<dovecot-bounces@dovecot.org> to=<miabuser@example.com> proto=ESMTP helo=<talvi.dovecot.org>
````

I emailed the list maintainer and he responded that they have not been able to keep off of the Spamhaus blacklists, despite the fact that there is no spam coming from their server.

I can appreciate how difficult it must be to come up with a configuration that works for everybody, but is it possible you might reconsider the rejection policy with regards to blacklists? IIRC, MiaB doesn't respect DMARC p=reject policies created by site admins, but you follow blacklists, which admins have zero control over.

**Comment by jvolkenant** [comment]
The issue is with spamhaus and not with MIAB.  You can try to [rbl whitelist](https://www.google.com/search?client=firefox-b-1-d&q=rbl+whitelist+postfix) this sender.  But your changes will be overwritten when you run miab install.

**Comment by myfirstnameispaul** [comment]
So, Spamhaus is configuring MiaB?

**Comment by JoshData** [comment]
Lol ok. Chill please.

**Comment by myfirstnameispaul** [comment]
I didn't intend to express anything other than this doesn't seem to me to be an "it's the other project's fault" type of an issue.

**Comment by pierreozoux** [comment]
@myfirstnameispaul as @jvolkenant said, you have to contact spamhaus. This project can't do anything to help you.

COuld you please close the issue? thanks!

**Comment by myfirstnameispaul** [comment]
@pierreozoux There is no option to send to spam instead of blocking?

**Comment by ctrl-i** [comment]
> @pierreozoux There is no option to send to spam instead of blocking?

As has already been recommended to you, you need to contact [Spamhaus](https://www.spamhaus.org/).  This is not something MIAB can do for you.  MIAB uses the Spamhaus lists to filter spam and MIAB has no way to edit the lists Spamhaus controls.

**Comment by gellenburg** [comment]
I'm having the same issue. I have several MIAB instances running at Linode. They are all in the New Jersey datacenter.

The MIAB instance I use for my personal Email and domain is not receiving mail sent from my own MIAB instance that I use for business email (and my business domain).

My personal MIAB instance is blocking email from my own domain due to zen.spamhaus.org.

Here's the rub. I know I have never sent any UCE from my business domain. Ever.

I have now gone through the process of getting my business MIAB removed from zen.spamhaus.org nine (yes **9**) separate times, and it stays removed for a week or so.

This is getting old.

We really need a configurable option to decide which blacklists we want to support. I'm almost to the point of forking MIAB just so I can remove any and all references to zen.spamhaus.org from all code and configurations.

Spamhaus is an utter and complete joke.

**Comment by jvolkenant** [comment]
The problem you will find however, is not that you have stopped checking for inbound mail with zen.spamhaus.org. It will be that the people you are emailing might be using zen.spamhaus.org.

**Comment by gellenburg** [comment]
Yep. Finding that out. Only recourse seems to be to convince folks to migrate off Miab since there's no other option.

Which sucks because I really hate iredmail. 😔

Get Outlook for Android<https://aka.ms/ghei36>
________________________________

**Comment by jvolkenant** [comment]
If you are listed on any blacklist; that is not the fault of Mail in a box (or any other email server)  I have bought VPS before that had IP's that were listed on blacklists that I could not get removed. I had to settle with changing IP's.

**Comment by gellenburg** [comment]
I'm not on any blacklist the problem is zen. I had a mail server that was in zen that I got removed then as a test I shut that box down I powered it off completely disabled the virtual machine then miraculously one week later Zen relisted that box again back in their blacklist The problem is zen.spamhaus.org and since mailing a box refuses to provide any sort of configurable option for disabling zen.spamhaus.org my only recourse was to migrate off of mail in a box which is what I'm doing as fast as I possibly can and educating others why they shouldn't be using mail in a box either I love mail in a box but the DNS block lists are hard coated and of course I can edit the post fixed main.cf file easy enough but the problem there is the next time I go to upgrade mail in a box that config file is going to get overwritten. I thought about forking mail in a box and creating my own project that didn't have zen.spamhaus.org but I don't have the time to keep something like that maintained. Am I frustrated Yes am I bitter a little bit because here's an organization that and it's only zen.spamhaus by the way that is causing me so much grief and frustration. apologies for the lack of formatting I'm on my phone and I'm using Google transcription.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by anoma** [comment]
Would you consider posting a redacted copy of your conversations with spamhaus as it seems you have identified a bug they are not addressing.

**Comment by gellenburg** [comment]
There is no way to contact spamhaus except through their website and filling out an automated form to get a server removed from the Zen block list.

Get Outlook for Android<https://aka.ms/ghei36>
________________________________

**Comment by anoma** [comment]
I have never had occasion to contact them but they seem to have a github presence at least. https://github.com/spamhaus

From reviewing the comments so far it would indeed seem they have a possible bug but without knowing IPs and domain names etc you will have to self verify that.

**Comment by gellenburg** [comment]
M i a b is blocking other email from other m i a b users because of Zen but more importantly the fact that male administrators can't control or don't have granular control over the spam policies that are applied to their installation is a design flaw of m i a b in my opinion.

Get Outlook for Android<https://aka.ms/ghei36>
________________________________

**Comment by myfirstnameispaul** [comment]
I am currently unable to receive an email sent to my MiaB installation because the sender uses a provider that is on a blacklist.

I would strongly prefer such an email go to my spam folder instead of being blocked by the server.

There is nothing to inform the sender of the problem, other than when I call them on the telephone.

**Comment by myfirstnameispaul** [comment]
Missed an important email from Citibank because their sending service was in the Spamhaus DBL.

I do not find this to be a useful feature.

**Comment by anoma** [comment]
It was remiss of Citibank not to act on this before you even knew about it.

RBLs are an imperfect but vital technology. For the handful of problem edge case we can report here millions, perhaps billions, of true spam mail is blocked to MIAB users because of it every year.

**Comment by gellenburg** [comment]
Not allowing a mail administrator granular control over the anti UCE policies applied to their organization's email is also remissive.

But we can't control citibank's policies.

Please allow us to control our's.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by JoshData** [comment]
I'm really getting tired of this thread. If you don't like things as they are you have SO MANY options:

* This product runs on your on server. Just modify the postfix settings on your server to your liking. But don't expect help here --- we're not volunteer sysadmins. You can probably find people willing to help with that elsewhere.
* This project is open source. You can propose an alternative, especially by submitting a pull request, for consideration. But as you can see from this thread some people find the blocklist to be useful (including me), so I will not accept a change that simply removes blocklists. You have to find a solution that satisfies concerns other than your own.
* If you think Mail-in-a-Box functionality wasn't sufficiently documented so that you knew what you were getting into before using it, write (or submit a pull request that adds) the documentation you wish existed.
* Modify the project for people who don't want blocklists and turn that into a new project --- use this project as the basis for your own product. (Forks are flattering. Just don't call it a Mail-in-a-Box --- for security reasons, it is important that people clearly know when they are and aren't using a product from this project.)
* Use a different product that aligns better with your requirements.

**Comment by gellenburg** [comment]
And if Mail In A Box didn’t overwrite my custom Postfix main.cf and other settings there wouldn’t be a problem, but every time MAIB is updated it wirtes out a new main.cf overwriting any customizations a user may have implemented.

 

That’s the crux of the problem. 😊

**Comment by JoshData** [comment]
I'm closing and locking this issue. If there are constructive proposals for improvements to Mail-in-a-Box, please go ahead and open a new issue for the proposal. I appreciate the feedback but I don't think any further discussion is going to add any new information here. Blocklists are sometimes going to block too much and I empathize when something important gets lost, but this is how Mail-in-a-Box works until someone comes up with something better.
